### ISSUE TITLE
apt update no longer works after latest updates

### ISSUE THREAD

**Issue by bentatham** [carlossg/docker-maven#283]
Prior to May 30, everything was fine in our extensions of the maven images (3.8.5-eclipse-temurin-11 specifically), but as of the latest updates (id 7fc2f17ef11d worked, but the 7278ed17f460 does not).

Specifically, the following fails on `apt update` in a Dockerfile using maven:3.8.5-eclipse-temurin-11 as the FROM.
```
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
```

This is likely related to an update of the base images, as this looks similar to [this issue](https://stackoverflow.com/questions/71941032/why-i-cannot-run-apt-update-inside-a-fresh-ubuntu22-04).  I also see that my previous builds updated from focal, and now it updates from jammy.  So on the same tag, the base image changed from Ubuntu 20 to 22, which surprised me.

**Comment by carlossg** [comment]
Looks the same as #281

The labels track upstream labels, so when they change distro and this image is rebuilt it gets updated

**Comment by bentatham** [comment]
Yes, this is similar to #281. I can confirm that using docker engine 20.10.15, the problem goes away.  (It was failing for us on Bitbucket Pipelines - but, via very efficient support from Bitbucket, they updated the docker engine to 20.10.15 and now our builds pass on the jammy-based images)

**Comment by Vortelf** [comment]
I just pulled `3-amazoncorretto-17-debian` to replace my OpenJDK image and encountered this bug. Using Docker `28.5.2`.

The reason for this is wrong permission when the one of the `apt` cache folders is created. All the other folders are created as root, while this one is created under wrong user and has no permissions outside of that user. 
`drwx------        42:0       /var/cache/apt/archives/partial`

<img width="2550" height="1354" alt="Image" src="https://github.com/user-attachments/assets/2bc5fe0b-9d33-4570-8f49-494a8e72c368" />
